Parser/rdiBeam2Earth.m

```diff
@@ -0,0 +1,194 @@
+function [veast,vnrth,wvel,evel]=rdiBeam2Earth(b1,b2,b3,b4,head,pitch,roll,ba,cvxccv,updown, distance)
+% Function to convert ADCP beam velocities to Earth coordinate velocity.
+% b1,b2,b3,b4 are four beam velocities
+% head is the heading reading, in degrees.
+% pitch is the pitch reading, in degrees.
+% roll is the roll reading, in degrees.
+% ba is the beam angle, in degrees.
+% cvxccv is 1 for a convex xdcr, other than 1 for a concave xdcr.  Usually 1.
+% updwon is 0 for a down-looking instrument, other than 0 for an up-looker.
+%
+% Calls matlab routines beam2inst.m and inst2earth.m.
+%
+head = head*0.01;
+pitch = pitch *0.01;
+roll = roll * 0.01;
+% make pitch, roll, heading the same size as b1, b2, b3, b4
+elev=[-70 -70 -70 -70]; % default elev for 20o
+azi=[270 90 0 180];
+if updown == 1 %uplooking
+    txtup = 'up';
+else
+    txtup = 'down';
+end
+[veast,vnrth,wvel,evel] = deal(NaN*b1);
+% have to bin map it first. Even if the EX bit is set for bin
+% mapping, if the data is collected in beam coordinates, bin mapping
+% does not occur on board.
+[b1,b2,b3,b4] = binmap(b1,b2,b3,b4,head,pitch,roll,ba,distance);
```

Comment:
better to stay with a single unit in the entire functionality. There are double conversion steps all over the place. Better to stick with radians from the start.

Reply:
Yes, whatever is easiest. As long as in the end, we can see the heading in degrees, I think.